inst/validation/val_report_child.Rmd

```diff
@@ -0,0 +1,205 @@
+tests[["spec_id_paste"]] <- vapply(tests[["spec_id"]], function(x) paste(x, collapse = "\n"), FUN.VALUE = character(1))
+tests[["spec_paste"]] <- vapply(tests[["spec"]], function(x) paste(x, collapse = "\n"), FUN.VALUE = character(1))
+tests[["desc"]] <- paste0("(#", seq_len(nrow(tests)), "): ", tests[["test"]])
+tests[["with_spec"]] <- vapply(tests[["spec_id"]], Negate(is.null), FUN.VALUE = logical(1))
+
+spec_tests <- tests[tests[["with_spec"]], ]
+no_spec_tests <- tests[!tests[["with_spec"]], ]
+
+declared_spec <- vdoc[["spec_id_list"]]
+tested_spec <- unique(unlist(tests[["spec_id"]]))
+uncovered_spec <- declared_spec[!declared_spec %in% tested_spec]
+undeclared_spec <- tested_spec[!tested_spec %in% declared_spec]
+
+spec_tests[["are_declared"]] <- sapply(spec_tests[["spec_id"]], function(x) all(x %in% declared_spec))
+
+# Count tests in the different categories ----
+
+mask_failed <- !!spec_tests[["failed"]]
+mask_skipped <- !!spec_tests[["skipped"]]
+mask_declared <- spec_tests[["are_declared"]]
+n_pass_dec <- sum(!mask_failed & !mask_skipped & mask_declared)
+n_fail_dec <- sum(mask_failed & mask_declared)
+n_skip_dec <- sum(mask_skipped & mask_declared)
+n_uncov <- length(uncovered_spec)
+n_undec <- sum(!mask_declared)
+
+render_spec_table <- function(t) {
+  t <- t[trac_matrix_col]
+  colnames(t) <- names(trac_matrix_col)
+  t <- t[order(t[["Spec ID"]]), ]
+  DT::datatable(t, options = list(dom = "ltp"), filter = list(position = "top"))
+}
+
+data_frame_by_row <- function(colnames, data) {
+  n <- length(data)
+  n_cols <- length(colnames)
+  stopifnot(n %% n_cols == 0)
+  columns <- vector("list", length = n_cols)
+  for (i in 1:n_cols) columns[[i]] <- unlist(data[seq(i, n, n_cols)])
+  do.call(data.frame, setNames(columns, colnames))
+}
+
+# Select columns to be included in the tables ----
+trac_matrix_col <- c("Spec ID" = "spec_id_paste", "Spec" = "spec_paste", "Test Desc" = "desc", "File" = "file")
+
+# Check that validation passes and set title ----
+VALIDATION_PASSED <- n_fail_dec == 0 && n_uncov == 0 && n_undec == 0 && n_uncov == 0 # nolint
+
+result_symbol <- if (VALIDATION_PASSED) "\U02705" else "\U274C"
+title <- paste(result_symbol, params[["package"]], params[["version"]])
+```
+
+## `r title`
+
+The following document generates a report for R packages, to satisfy the criteria of a "Released" status under the **Non-GxP** project. The QC report contains the following information:
+
+- **Specifications (specs):** These can be attached to every test that the user adds.
+- **Traceability matrix:** Contains test cases with passed, failed, or skipped expectations.
+- **Uncovered or undeclared specs**
+- **Session Info and System Configuration**
+
+::: {.infobox .warning}
+Please be advised that the QC report generated for this module does not imply validation according to any other GxP criteria. 
+The QC report only satisfies our internally developed quality checks for non-GxP criteria.
```

Comment:
disambiguate "our" here.

Reply:
Agreed that qc report requires a review to match the contents. I don't think that the prologue is 100% accurate. Nonetheless, the contents should be further discussed internally and I don't think this should be covered in this PR.